### DISCUSSION TITLE
OnUserAfterVerification

### DISCUSSION THREAD

**Discussion by KyungWonPark** [pocketbase/pocketbase#727] [Q&A] [answered]
Hello

Could you make "OnUserAfterVerification" event hook?
This would come handy when you need to add user to different service(ex; create linux user) but only when they are verified through email

thank you

**Comment by ganigeorgiev** [comment, marked as answer]
It could be added but that will be after v0.8. You could watch issue https://github.com/pocketbase/pocketbase/issues/468.

As a workaround for now you can use the `OnModelBeforeUpdate` hook:

```go
app.OnModelBeforeUpdate().Add(func(e *core.ModelEvent) error {
    if e.Model.TableName() != "_users" {
        return nil // not a user model
    }

    user, _ := e.Model.(*models.User)

    // in the future the old state will be part of the event
    oldUser, err := app.Dao().FindUserById(user.Id)
    if err != nil {
        return err
    }

    if user.Verified && oldUser.Verified != user.Verified {
        // your custom call...
    }

    return nil
})
```
